test/e2e/helloworld_shell_test.go

```diff
@@ -106,6 +106,15 @@ func TestHelloWorldFromShell(t *testing.T) {
 
 	logger.Info("Waiting for ingress to come up")
 	gateway := "istio-ingressgateway"
+	gatewayNamespace := "istio-system"
+
+	if gatewayOverride := os.Getenv("GATEWAY_OVERRIDE"); gatewayOverride != "" {
+		gateway = gatewayOverride
+	}
+	if gatewayNsOverride := os.Getenv("GATEWAY_NAMESPACE_OVERRIDE"); gatewayNsOverride != "" {
+		gatewayNamespace = gatewayNsOverride
+	}
```

Comment:
Can we make these test flags to pass to the tests? We use flags to override values elsewhere and we could just add `-gateway` and `-gatewayNamespace` (without override) and default them to the istio ones for now. WDYT?

Reply:
+1

Reply:
fixed, but note that https://github.com/knative/pkg/blob/master/test/spoof/spoof.go#L133 still requires these env vars to be set